pkg/agent/openflow/network_policy.go

```diff
@@ -2220,3 +2228,64 @@ func (f *featureNetworkPolicy) l7NPTrafficControlFlows() []binding.Flow {
 			Done(),
 	}
 }
+
+func (f *featureNetworkPolicy) loggingNPPacketFlowWithOperations(cookieID uint64, operations uint8) binding.Flow {
+	loggingOperations := binding.NewRegMark(PacketInOperationField, uint32(operations))
+	fb := L2ForwardingOutTable.ofTable.BuildFlow(priorityNormal).
+		MatchRegMark(SendToControllerRegMark, loggingOperations)
+	if f.ovsMetersAreSupported {
+		fb = fb.Action().Meter(PacketInMeterIDNP)
+	}
+	return fb.Action().SendToController([]byte{uint8(PacketInCategoryNP), operations}, false).
+		Cookie(cookieID).
+		Done()
+}
+
+func (f *featureNetworkPolicy) initLoggingFlows() []binding.Flow {
+	flows := make([]binding.Flow, 0)
```

Comment:
nit: `var flows []binding.flow`, and initialize its len to `PacketInNPLoggingOperation+PacketInNPStoreDenyOperation+PacketInNPRejectOperation` below

Reply:
added